x-pack/plugins/data_enhanced/server/search/es_search_strategy.ts

```diff
@@ -112,20 +100,27 @@ async function asyncSearch(
   // Only report partial results every 64 shards; this should be reduced when we actually display partial results
   const batchedReduceSize = request.id ? undefined : 64;
 
-  const query = toSnakeCase({
+  const asyncOptions = {
     waitForCompletionTimeout: '100ms', // Wait up to 100ms for the response to return
     keepAlive: '1m', // Extend the TTL for this search request by one minute
+  };
+
+  const querystring = toSnakeCase({
+    ...asyncOptions,
     ...(batchedReduceSize && { batchedReduceSize }),
     ...queryParams,
   });
 
-  // eslint-disable-next-line @typescript-eslint/naming-convention
-  const { id, response, is_partial, is_running } = (await caller(
-    'transport.request',
-    { method, path, body, query },
-    options
-  )) as AsyncSearchResponse<any>;
+  // TODO: replace with async endpoints once https://github.com/elastic/elasticsearch-js/issues/1280 is resolved
+  const esResponse = await client.transport.request({
+    method,
+    path,
+    body,
+    querystring,
+  });
 
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  const { id, response, is_partial, is_running } = esResponse.body;
```

Comment:
nit: You can get rid of this es-lint-disable line if you do something like this:
```suggestion
  const { id, response, is_partial: isPartial, is_running: isRunning } = esResponse.body;
```